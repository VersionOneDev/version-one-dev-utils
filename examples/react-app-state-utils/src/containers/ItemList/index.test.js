// React
import React from "react";
// Testing
import { render, screen, waitFor } from "@testing-library/react";
import { composeStories } from "@storybook/testing-react";
import userEvent from "@testing-library/user-event";
// Stories
import * as stories from "./index.stories";

const { Template } = composeStories(stories);

global.STORYBOOK_ACTION = jest.fn();

/** Template story */
describe("ItemList (Template Story)", () => {
  it("Matches snapshot", () => {
    expect(render(<Template />)).toMatchSnapshot();
  });

  it("Changes route to item when view is clicked", () => {
    render(<Template />);
    userEvent.click(screen.getByTestId("ItemList/itemView/1"));
    expect(global.STORYBOOK_ACTION).toHaveBeenCalledWith("/item/1");
  });

  it("Displays Heading component", () => {
    render(<Template />);
    expect(screen.getByTestId("ItemList/title"));
  });

  it("Fetches items from ItemStore", async () => {
    render(<Template />);
    await waitFor(() =>
      expect(global.STORYBOOK_ACTION).toHaveBeenCalledWith({
        type: "ItemStore/get",
      })
    );
  });
});
