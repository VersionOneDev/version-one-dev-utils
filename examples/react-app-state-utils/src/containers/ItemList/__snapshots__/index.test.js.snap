// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList (Template Story) Matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="mb-10 text-4xl text-center"
        data-testid="ref/title"
      >
        To Do List
      </h1>
      <ul>
        <li
          class="p-6 mb-4 block rounded-md bg-gray-400 hover:bg-gray-300 flex items-center cursor-pointer bg-gray-400 hover:bg-gray-500"
        >
          <div
            class="flex-1 mr-4 text-gray-700"
          >
            <span
              class="line-through"
              data-testid="ref/itemTitle/1"
            >
              Title!!
            </span>
            <div>
              <a
                class="underline hover:no-underline text-xs mt-1"
                data-testid="ref/itemView/1"
                href="/item/1"
              >
                View
              </a>
            </div>
          </div>
          <span
            class="inline-block w-6 h-6 border-2 border-gray-600 rounded-full border-green-700 bg-green-400"
          />
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <h1
      class="mb-10 text-4xl text-center"
      data-testid="ref/title"
    >
      To Do List
    </h1>
    <ul>
      <li
        class="p-6 mb-4 block rounded-md bg-gray-400 hover:bg-gray-300 flex items-center cursor-pointer bg-gray-400 hover:bg-gray-500"
      >
        <div
          class="flex-1 mr-4 text-gray-700"
        >
          <span
            class="line-through"
            data-testid="ref/itemTitle/1"
          >
            Title!!
          </span>
          <div>
            <a
              class="underline hover:no-underline text-xs mt-1"
              data-testid="ref/itemView/1"
              href="/item/1"
            >
              View
            </a>
          </div>
        </div>
        <span
          class="inline-block w-6 h-6 border-2 border-gray-600 rounded-full border-green-700 bg-green-400"
        />
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
